main.js

```diff
@@ -194,18 +203,81 @@ function createWindow () {
   });
 }
 
+function createTrayIcon() {
+  logger.info('Creating tray icon');
+
+  // TODO: better to not use a hardcoded icon file
+  tray = new Tray(path.join(__dirname, '/images/icon_48.png'));
+
+  trayContextMenu = Menu.buildFromTemplate([
+      {id: 'show', label: locale.messages.show.message, visible: !mainWindow.isVisible(), click: () => {showWindow();}},
+      {id: 'hide', label: locale.messages.hide.message, visible: mainWindow.isVisible(), click: () => {hideWindow();}},
+      {id: 'quit', label: locale.messages.quit.message, click: () => {app.quit();}}
+  ]);
+
+  // NOTE: a function equivalent to the following will become part of the
+  // prototype of Menu in a future release of electron (atm it is in 1.8.1-beta),
+  // so this definition can be removed in the future, when that version of electron
+  // will be released and stable. For details, please see:
+  // https://github.com/electron/electron/releases/tag/v1.8.1
+  // https://github.com/electron/electron/pull/10618
+  trayContextMenu.getMenuItemById = function (id) {
+    return this.items.find(item => item.id === id) || null;
+  }
+
+  tray.setContextMenu(trayContextMenu);
+
+  tray.on('clicked', () => {toggleWindowVisibility();});
```

Comment:
What's the difference between the 'clicked' and 'click' events?

Also, another situation where we can dispense with the surrounding function.

Reply:
Whoops. 'click' is the event. 'clicked' was an experiment I made, and somehow I missed to remove that line later.